timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -936,6 +936,56 @@ def evaluate(
             )
         return scores_dict
 
+    def get_feature_importance(
+        self,
+        data: TimeSeriesDataFrame,
+        features: List[str],
+        feature_importance_transform: AbstractFeatureImportanceTransform,
+        model: Optional[Union[str, AbstractTimeSeriesModel]] = None,
+        metric: Optional[Union[str, TimeSeriesScorer]] = None,
+        time_limit: Optional[float] = None,
+        subsample_size: int = 50,
+        num_iterations: int = 1,
+    ) -> pd.DataFrame:
+        metric = check_get_evaluation_metric(metric) if metric else self.eval_metric
+
+        logger.info("Computing feature importance")
+
+        if len(features) == 0:
+            raise ValueError(
+                "No features provided to compute feature importance. At least some valid features should be provided."
+            )
+
+        time_start = time.time()
+        if subsample_size > data.num_items:
+            logger.info(
+                f"Subsample_size {subsample_size} is larger than the number of items in the data and will be ignored"
+            )
+            subsample_size = data.num_items
+
+        relative_drops = defaultdict(list)
+        for n in range(num_iterations):
+            if subsample_size < data.num_items:
+                item_ids_sampled = data.item_ids.to_series().sample(subsample_size)
+                data_sample = data.loc[item_ids_sampled]
```

Comment:
`data.query("item_id in @item_ids_sampled")` is for some reason much faster than `loc` for large dataframes